GitUI/CommandsDialogs/BrowseDialog/GitStatusMonitor.cs

```diff
@@ -384,86 +305,102 @@ private void Update()
+                // repository status will change after these actions.
                 UICommandsSource.UICommands.RepoChangedNotifier.IsLocked ||
                 (GitVersion.Current.RaceConditionWhenGitStatusIsUpdatingIndex && Module.IsRunningGitProcess()))
             {
-                _statusIsUpToDate = false; // tell that computed status isn't up to date
+                _statusIsUpToDate = false; // schedule new update when command is finished
                 return;
             }
 
+            _workTreeWatcher.EnableRaisingEvents = true;
             _commandIsRunning = true;
             _statusIsUpToDate = true;
             _previousUpdateTime = Environment.TickCount;
 
+            // Schedule update every 5 min, even if we don't know that anything changed
+            CalculateNextUpdateTime(PeriodicUpdateInterval);
+
+            // capture a consistent state in the main thread
+            IGitModule module = Module;
+
             ThreadHelper.JoinableTaskFactory.RunAsync(
                 async () =>
                 {
                     try
                     {
-                        await TaskScheduler.Default;
-
-                        _ignoredFilesPending = _ignoredFilesAreStale;
+                        try
+                        {
+                            await TaskScheduler.Default;
 
-                        // git-status with ignored files when needed only
-                        var cmd = GitCommandHelpers.GetAllChangedFilesCmd(!_ignoredFilesPending, UntrackedFilesMode.Default, noLocks: true);
-                        var output = Module.RunGitCmd(cmd);
-                        var changedFiles = GitCommandHelpers.GetStatusChangedFilesFromString(Module, output);
+                            var cmd = GitCommandHelpers.GetAllChangedFilesCmd(true, UntrackedFilesMode.Default, noLocks: true);
+                            var output = module.RunGitCmd(cmd);
+                            var changedFiles = GitCommandHelpers.GetStatusChangedFilesFromString(module, output);
 
-                        await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                        UpdatedStatusReceived(changedFiles);
+                            if (!ModuleHasChanged())
+                            {
+                                UpdatedStatusReceived(changedFiles);
+                            }
+                        }
+                        catch
+                        {
+                            try
+                            {
+                                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                                // Avoid possible popups on every file changes
+                                CurrentStatus = GitStatusMonitorState.Stopped;
+                            }
+                            finally
```

Comment:
Did you mean `catch`? I see no point in having try-finally with an empty `finally` block.